sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -116,6 +116,10 @@ object ColumnPruning extends Rule[LogicalPlan] {
     case a @ Aggregate(_, _, child) if (child.outputSet -- a.references).nonEmpty =>
       a.copy(child = Project(a.references.toSeq, child))
 
+    case p @ Project(pl, Aggregate(ge, ec, child))
```

Comment:
I'd follow the convention from the rest of the optimizer and use longer variable names (i.e. `projectList`, `groupExprs`, `aggExprs`)